src/module.ts

```diff
@@ -0,0 +1,136 @@
+import { Handlers as SentryHandlers, captureException, withScope } from '@sentry/node'
+import type { ModuleConfiguration } from '../types'
+import type { DeepPartialModuleConfiguration } from '../types/sentry'
+import { envToBool, boolToText, canInitialize, clientSentryEnabled, serverSentryEnabled } from './core/utils'
+import { buildHook, initializeServerSentry, shutdownServerSentry, webpackConfigHook } from './core/hooks'
+
+export interface ModuleOptions extends DeepPartialModuleConfiguration {}
+
+export interface ModulePublicRuntimeConfig extends DeepPartialModuleConfiguration {}
+
+export interface ModulePrivateRuntimeConfig extends DeepPartialModuleConfiguration {}
+
+const logger = useLogger('nuxt:sentry')
+
+export default defineNuxtModule<ModuleConfiguration>({
+  meta: {
+    name: '@nuxtjs/sentry',
+    configKey: 'sentry',
+    compatibility: {
+      nuxt: '^2.15.8',
+    },
+  },
+  defaults: nuxt => ({
+    lazy: false,
+    dsn: process.env.SENTRY_DSN || '',
+    disabled: envToBool(process.env.SENTRY_DISABLED) || false,
+    initialize: envToBool(process.env.SENTRY_INITIALIZE) || true,
+    runtimeConfigKey: 'sentry',
+    disableClientSide: envToBool(process.env.SENTRY_DISABLE_CLIENT_SIDE) || false,
+    disableServerSide: envToBool(process.env.SENTRY_DISABLE_SERVER_SIDE) || false,
+    publishRelease: envToBool(process.env.SENTRY_PUBLISH_RELEASE) || false,
+    disableServerRelease: envToBool(process.env.SENTRY_DISABLE_SERVER_RELEASE) || false,
+    disableClientRelease: envToBool(process.env.SENTRY_DISABLE_CLIENT_RELEASE) || false,
+    logMockCalls: true,
+    sourceMapStyle: 'source-map',
+    tracing: false,
+    clientIntegrations: {
+      Dedupe: {},
+      ExtraErrorData: {},
+      ReportingObserver: {},
+      RewriteFrames: {},
+      Vue: { attachProps: true, logErrors: nuxt.options.dev },
+    },
+    serverIntegrations: {
+      Dedupe: {},
+      ExtraErrorData: {},
+      RewriteFrames: {},
+      Transaction: {},
+    },
+    customClientIntegrations: '',
+    customServerIntegrations: '',
+    config: {
+      environment: nuxt.options.dev ? 'development' : 'production',
+    },
+    serverConfig: {},
+    clientConfig: {},
+    requestHandlerConfig: {},
+  }),
+  setup (options, nuxt) {
+    const defaultsPublishRelease: SentryCliPluginOptions = {
+      include: [],
+      ignore: [
+        'node_modules',
+        '.nuxt/dist/client/img',
+      ],
+      configFile: '.sentryclirc',
+    }
+
+    if (options.publishRelease) {
+      options.publishRelease = defu(options.publishRelease, defaultsPublishRelease)
+    }
+
+    if (serverSentryEnabled(options)) {
+      // @ts-ignore
+      nuxt.hook('render:setupMiddleware', app => app.use(SentryHandlers.requestHandler(options.requestHandlerConfig)))
+      // @ts-ignore
+      nuxt.hook('render:errorMiddleware', app => app.use(SentryHandlers.errorHandler()))
+      // @ts-ignore
+      nuxt.hook('generate:routeFailed', ({ route, errors }) => {
```

Comment:
What's the deal with those hooks in Nuxt 3? Are those flat-out not supported (in which case how would that code work?) or not officially supported but still working? Or does this only work in Nuxt 2?

(Type ignoring is obviously not ideal so I would like to know the reason for that)

Reply:
First of all this is just the first step to make it compatible with Nuxt 3 it is not right now. And as for my understanding of the hooks situation I think it is as you said and they only work on Nuxt 2 and Nuxt 3 just ignores them.

The type ignoring is due to using Nuxt 3 types and those hooks not being present for Nuxt 3 🙈

Reply:
If this is only the first step then I guess it would make sense to merge it into a `next` branch or something rather than to `main`. Or do you mean to continue working on it in this PR?

Reply:
A next branch would be better. My goal for this PR was to prepare the path to add the Nuxt 3 compatibility without breaking the Nuxt 2 compatibility